nupic/torch/functions/k_winners.py

```diff
@@ -188,3 +189,74 @@ def backward(ctx, grad_output):
         grad_x = grad_x.reshape(grad_output.shape)
 
         return grad_x, None, None, None
+
+
+class KWinners2dLocal(torch.autograd.Function):
+    """
+    A K-winner take all autograd function for CNN 2D inputs (batch, Channel, H, W)
+    where the k-winners are chosen locally across all the channels
```

Comment:
`A K-winner take all autograd function for CNN 2D inputs (batch, Channel, H, W)
where k-winners are chosen independently for each location. Local k-winners selects the top k channels locally for each of the H X W locations.`